cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -300,10 +300,12 @@ bool SubstraitToVeloxPlanValidator::validateCast(
     case TypeKind::VARBINARY:
       LOG_VALIDATION_MSG("Invalid input type in casting: ARRAY/MAP/ROW/VARBINARY.");
       return false;
-    case TypeKind::TIMESTAMP: {
-      LOG_VALIDATION_MSG("Casting from TIMESTAMP is not supported or has incorrect result.");
-      return false;
-    }
+    case TypeKind::TIMESTAMP:
+      // Only support cast timestamp to date
+      if (!toType->isDate()) {
+        LOG_VALIDATION_MSG("Casting from TIMESTAMP is not supported or has incorrect result.");
```

Comment:
Can we also tell the target type in the message? i.e., it's better to include `toType->toString()`.

Reply:
updated, thank you.